### PR TITLE
AuthenticationService_CheckToken

### PR DESCRIPTION
**Pull request by djino97**
- Added user jwt validation logic for check token.
- Added command pattern ValidationUserJwtCommand.
- Added consumer for read from broker queue UserJwtConsumer.
- At the moment unit-test are not added, but it will added soon.

### CODE REVIEW

File: DigitalOffice/Services/AuthenticationService/Token/Interfaces/IJwtSigningEncodingKey.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    public interface IJwtSigningEncodingKey
+    {
+        ///<value>Type of algorithm encoding(HS256).</value>
+        string SigningAlgorithm { get; }
+
+        /// <summary>
+        ///Method for getting encoding key jwt.
+        /// </summary>
+        /// <returns>Return key to create the signature(private key).</returns>
+
```
Comment (KostinP) [inline DigitalOffice/Services/AuthenticationService/Token/Interfaces/IJwtSigningEncodingKey.cs]: Extra empty line.

File: DigitalOffice/Services/AuthenticationService/Token/Interfaces/IJwtSigningEncodingKey.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    /// <summary>
+    /// Represents interface for create encoding key jwt.
+    /// </summary>
+    public interface IJwtSigningEncodingKey
+    {
+        ///<value>Type of algorithm encoding(HS256).</value>
+        string SigningAlgorithm { get; }
+
+        /// <summary>
+        ///Method for getting encoding key jwt.
```
Comment (KostinP) [inline DigitalOffice/Services/AuthenticationService/Token/Interfaces/IJwtSigningEncodingKey.cs]: Not enough space.

File: DigitalOffice/Services/AuthenticationService/Token/Interfaces/IJwtValidation.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+﻿namespace LT.DigitalOffice.AuthenticationService.Token.Interfaces
+{
+    public interface IJwtValidation
```
Comment (KostinP) [inline DigitalOffice/Services/AuthenticationService/Token/Interfaces/IJwtValidation.cs]: XML?

File: DigitalOffice/Shared/Kernel/Broker/OperationResultWrapper.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace LT.DigitalOffice.Kernel.Broker
+{
+    /// <summary>
+    /// Wraps the returned object from a function in OperationResult.
+    /// </summary>
+    /// <typeparam name="T">parameter of the called function</typeparam>
+    /// <typeparam name="TRes">response from the called function, which will be wrapped in OperationResult</typeparam>
```
Comment (KostinP) [inline DigitalOffice/Shared/Kernel/Broker/OperationResultWrapper.cs]: Add dots.

File: DigitalOffice/Tests/AuthenticationServiceTests/Token/JwtValidationTests.cs [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+using System;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+using Microsoft.IdentityModel.Tokens;
+using LT.DigitalOffice.AuthenticationService.Token;
+
+namespace LT.DigitalOffice.AuthenticationServiceUnitTests.Token
+{
+
```
Comment (KostinP) [inline DigitalOffice/Tests/AuthenticationServiceTests/Token/JwtValidationTests.cs]: Extra empty line.

**Review by KostinP — APPROVED** [review]

File: DigitalOffice/Services/AuthenticationService/Token/JwtValidation.cs [hunk 6]
```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.IdentityModel.Tokens.Jwt;
+using LT.DigitalOffice.AuthenticationService.Token.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.IdentityModel.Tokens;
+
+namespace LT.DigitalOffice.AuthenticationService.Token
+{
+    public class JwtValidation : IJwtValidation
```
Comment (MihailKulikov) [inline DigitalOffice/Services/AuthenticationService/Token/JwtValidation.cs]: JwtValidator will be better (not nessesary)

File: DigitalOffice/Services/AuthenticationService/Token/JwtValidation.cs [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        {
+            this.validationParameters = validationParameters;
+        }
+
+        public void ValidateJwt(string jwt)
+        {
+            try
+            {
+                new JwtSecurityTokenHandler()
+                    .ValidateToken(jwt, validationParameters, out var validatedToken);
```
Comment (MihailKulikov) [inline DigitalOffice/Services/AuthenticationService/Token/JwtValidation.cs]: Unused local variable validatedToken

File: DigitalOffice/Services/AuthenticationService/Token/JwtValidation.cs [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+            }
+            catch(SecurityTokenValidationException exception)
+            {
+                throw new Exception($"Token failed validation: {exception.Message}");
+            }
+            catch(ArgumentException exception)
+            {
+                var a = exception.Message;
```
Comment (MihailKulikov) [inline DigitalOffice/Services/AuthenticationService/Token/JwtValidation.cs]: Unused

File: DigitalOffice/Tests/AuthenticationServiceTests/Broker/Consumers/UserJwtConsumerTests.cs [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+using LT.DigitalOffice.Broker.Requests;
+using LT.DigitalOffice.Kernel.Broker;
+using MassTransit.Testing;
+using Moq;
+using NUnit.Framework;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.UserServiceUnitTests.Broker.Consumer
```
Comment (MihailKulikov) [inline DigitalOffice/Tests/AuthenticationServiceTests/Broker/Consumers/UserJwtConsumerTests.cs]: fix namespace

**Review by lalagvana — APPROVED** [review]

File: DigitalOffice/Services/AuthenticationService/Token/Interfaces/IJwtSigningEncodingKey.cs (hunk 1 above)
Comment (djino97) [reply DigitalOffice/Services/AuthenticationService/Token/Interfaces/IJwtSigningEncodingKey.cs]: Fixed

File: DigitalOffice/Services/AuthenticationService/Token/Interfaces/IJwtSigningEncodingKey.cs (hunk 2 above)
Comment (djino97) [reply DigitalOffice/Services/AuthenticationService/Token/Interfaces/IJwtSigningEncodingKey.cs]: Fixed

File: DigitalOffice/Services/AuthenticationService/Token/Interfaces/IJwtValidation.cs (hunk 3 above)
Comment (djino97) [reply DigitalOffice/Services/AuthenticationService/Token/Interfaces/IJwtValidation.cs]: Fixed

File: DigitalOffice/Shared/Kernel/Broker/OperationResultWrapper.cs (hunk 4 above)
Comment (djino97) [reply DigitalOffice/Shared/Kernel/Broker/OperationResultWrapper.cs]: Fixed

File: DigitalOffice/Tests/AuthenticationServiceTests/Token/JwtValidationTests.cs (hunk 5 above)
Comment (djino97) [reply DigitalOffice/Tests/AuthenticationServiceTests/Token/JwtValidationTests.cs]: Fixed

File: DigitalOffice/Services/AuthenticationService/Token/JwtValidation.cs (hunk 7 above)
Comment (djino97) [reply DigitalOffice/Services/AuthenticationService/Token/JwtValidation.cs]: I writed _, for this function parametr

File: DigitalOffice/Services/AuthenticationService/Token/JwtValidation.cs (hunk 8 above)
Comment (djino97) [reply DigitalOffice/Services/AuthenticationService/Token/JwtValidation.cs]: Fixed

File: DigitalOffice/Tests/AuthenticationServiceTests/Broker/Consumers/UserJwtConsumerTests.cs (hunk 9 above)
Comment (djino97) [reply DigitalOffice/Tests/AuthenticationServiceTests/Broker/Consumers/UserJwtConsumerTests.cs]: Fixed

File: DigitalOffice/Services/AuthenticationService/Token/JwtValidation.cs (hunk 6 above)
Comment (djino97) [reply DigitalOffice/Services/AuthenticationService/Token/JwtValidation.cs]: Fixed

**Review by MihailKulikov — APPROVED** [review]

**Review by mekk1t — APPROVED** [review]

**Review by ifelseelif — APPROVED** [review]

**Pull request merged**